crates/tako/src/internal/worker/rqueue.rs

```diff
@@ -67,6 +127,15 @@ impl ResourceWaitQueue {
         task_map: &TaskMap,
         remaining_time: Option<Duration>,
     ) -> Vec<(TaskId, Allocation)> {
+        if self.need_recompute_resource_priorities {
```

Comment:
Maybe we could do this when workers are added/removed and get rid of the bool? It would be slightly amortized here, but I would rather not put computation into the latency critical task starting when it's not needed. Worker connect/disconnect should be rare enough that we could just recompute it every time.